tldw_Server_API/tests/Web_Scraping/test_http_client_fetch.py

```
@@ -131,3 +138,322 @@ def get(self, url, **kwargs):
+    assert [url for url, _ in calls] == [
+        "https://example.com/start",
+        "https://example.com/final",
+    ]
+    assert calls[0][1]["allow_redirects"] is False
+    assert calls[1][1]["allow_redirects"] is False
+    assert resp["status"] == 200
+    assert resp["url"] == "https://example.com/final"
+
+
+def test_curl_fetch_follows_same_host_redirects_without_httpx(monkeypatch):
+    monkeypatch.setattr(hc, "_is_url_allowed", lambda url: True)
```

Ruff (0.15.9)

[warning] Unused lambda argument: `url`

(ARG005)